vpr/src/place/initial_placement.cpp

```diff
@@ -1204,17 +1204,22 @@ void initial_placement(const t_placer_opts& placer_opts,
         read_constraints(constraints_file);
     }
 
-    if (noc_opts.noc) {
-        // NoC routers are placed before other blocks
-        initial_noc_placement(noc_opts, placer_opts);
-        propagate_place_constraints();
-    }
+    if(!placer_opts.read_initial_place_file.empty()) {
+        const auto& grid = g_vpr_ctx.device().grid;
+        read_place(nullptr, placer_opts.read_initial_place_file.c_str(), false, grid);
```

Comment:
What happens if both constraint and initial placement files are specified?

I think read_constraints() initializes the location of some blocks and mark them as fixed. The location in the initial placement would override the location in the constraint file. We probably need to handle this case

Reply:
I think it is better to print something to show that the initial placement is read from a file.

Reply:
You are right. The locations in the initial placement file will override the assigned locations in the constraint file. However, at the end of the initial placement, we call a function to check the legality of the placement, and any issues can be caught there.

Regarding printing that the initial placement is read from a file, the `read_place` function prints a message indicating that the placement is read from a file, along with the file name.

Reply:
Are you referring to `check_initial_placement_legality()` or another function?

Reply:
Yes, I meant this function. I thought it would check whether the location of the fixed blocks matched the one specified in the constraint file, but upon reviewing the implementation, I realized that's not the case.

To avoid slowing down `set_block_location`, which is on the critical path of the placement code, I added the check when the placement file is read in the `read_place.cpp` file.